packages/apps/proof-of-us/src/hooks/account.hook.tsx

```diff
@@ -0,0 +1,91 @@
+'use client';
+import { ACCOUNTCOOKIENAME } from '@/constants';
+import { env } from '@/utils/env';
+import { useRouter, useSearchParams } from 'next/navigation';
+import type { FC, PropsWithChildren } from 'react';
+import {
+  createContext,
+  useCallback,
+  useContext,
+  useEffect,
+  useState,
+} from 'react';
+import { useToasts } from './toast.hook';
+
+interface IAccountError {
+  message: string;
+}
+
+interface IAccountContext {
+  account?: IAccount;
+  error?: IAccountError;
+  login: () => void;
+  logout: () => void;
+}
+
+export const AccountContext = createContext<IAccountContext>({
+  account: undefined,
+  login: () => {},
+  logout: () => {},
+});
+
+export const useAccount = (): IAccountContext => useContext(AccountContext);
+
+export const AccountProvider: FC<PropsWithChildren> = ({ children }) => {
+  const [account, setAccount] = useState<IAccount>();
+  const { addToast } = useToasts();
+  const router = useRouter();
+  const searchParams = useSearchParams();
+
```

Comment:
This more seems a provider component rather than a hook. So, imo you could split this file into two maybe one for `context` and another one as a separate provider component.